airflow/timetables/events.py

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Iterable, Optional
+
+import numpy as np
+import pendulum
+from pendulum import DateTime
+
+from airflow.timetables.base import DagRunInfo, DataInterval, TimeRestriction, Timetable
+from airflow.timetables.simple import NullTimetable
+
+
+class EventsTimetable(NullTimetable):
```

Comment:
`NullTimetable` (and other non-recurring timetables) has some special treatments in UI. I think it’s better to inherit `Timetable` instead (the root class). Also it should be useful to implement `__repr__` and `summary` For UI representation.

Reply:
Changed to inherit from Timetable and added as simple text for all the `description`, `summary`, and `__repr__`.